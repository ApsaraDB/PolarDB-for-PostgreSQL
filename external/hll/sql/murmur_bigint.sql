-- ================================================================
-- Setup the table
--

SELECT hll_set_output_version(1);

DROP TABLE IF EXISTS test_seznjqbb;

CREATE TABLE test_seznjqbb (
    recno                       SERIAL,
    seed                        integer,
    pre_hash_long               bigint,
    post_hash_long              bigint
);

\copy test_seznjqbb (seed, pre_hash_long, post_hash_long) from sql/data/murmur_bigint.csv with csv header

SELECT COUNT(*) FROM test_seznjqbb;

SELECT recno, post_hash_long, hll_hash_bigint(pre_hash_long, seed)
  FROM test_seznjqbb
 WHERE hll_hashval(post_hash_long) != hll_hash_bigint(pre_hash_long, seed)
 ORDER BY recno;

DROP TABLE test_seznjqbb;
