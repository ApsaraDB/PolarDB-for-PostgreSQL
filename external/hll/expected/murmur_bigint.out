-- ================================================================
-- Setup the table
--
SELECT hll_set_output_version(1);
 hll_set_output_version 
------------------------
                      1
(1 row)

DROP TABLE IF EXISTS test_seznjqbb;
NOTICE:  table "test_seznjqbb" does not exist, skipping
CREATE TABLE test_seznjqbb (
    recno                       SERIAL,
    seed                        integer,
    pre_hash_long               bigint,
    post_hash_long              bigint
);
\copy test_seznjqbb (seed, pre_hash_long, post_hash_long) from sql/data/murmur_bigint.csv with csv header
SELECT COUNT(*) FROM test_seznjqbb;
 count 
-------
   600
(1 row)

SELECT recno, post_hash_long, hll_hash_bigint(pre_hash_long, seed)
  FROM test_seznjqbb
 WHERE hll_hashval(post_hash_long) != hll_hash_bigint(pre_hash_long, seed)
 ORDER BY recno;
 recno | post_hash_long | hll_hash_bigint 
-------+----------------+-----------------
(0 rows)

DROP TABLE test_seznjqbb;
